spec/controllers/prompts_controller_spec.rb

```diff
@@ -1,18 +1,127 @@
-require 'spec_helper'
+require "spec_helper"
 
 describe PromptsController do
   include LoginMacros
+  include RedirectExpectationHelper
+  let(:collection) { create(:collection) }
+  let(:open_signup) do
+    signups = create(:challenge_signup)
+    signups.collection.challenge.signup_open = true
+    signups.collection.challenge.save
+    signups
+  end
+  let(:signup) { create(:challenge_signup) }
 
+  let(:user) { create(:user) }
   before do
-    fake_login
+    fake_login_known_user(user)
+  end
+
+  describe "no_challenge" do
+    it "should show an error and redirect" do
+      get :show, collection_id: collection.name
+      it_redirects_to_with_error(collection_path(collection), "What challenge did you want to sign up for?")
+    end
+  end
+
+  describe "no_signup" do
+    it "should show an error and redirect" do
+      post :create, collection_id: signup.collection.name
+      it_redirects_to_with_error(collection_signups_path(signup.collection) + "/new",\
+                                 "Please submit a basic sign-up with the required fields first.")
+    end
+  end
+
+  describe "signups_closed" do
+    let(:user) { Pseud.find(ChallengeSignup.in_collection(signup.collection).first.pseud_id).user }
+    it "should show an error and redirect" do
+      post :create, collection_id: signup.collection.name
+      it_redirects_to_with_error(signup.collection, "Signup is currently closed: please contact a moderator for help.")
+    end
+  end
+
+  describe "not_signup_owner" do
+    it "should show an error and redirect" do
+      prompt = signup.collection.prompts.first
+      post :edit, id: prompt.id, collection_id: signup.collection.name
+      it_redirects_to_with_error(signup.collection, "You can't edit someone else's sign-up!")
+    end
+  end
+
+  describe "new" do
+    context "when prompt_type is offer" do
+      it "should have no errors" do
+        # Login as the signup owner
+        fake_login_known_user(Pseud.find(ChallengeSignup.in_collection(open_signup.collection).first.pseud_id).user)
```

Comment:
@annelu I did try for example:

  describe "new" do
    context "when prompt_type is offer" do
      let(:user) { Pseud.find(ChallengeSignup.in_collection(signup.collection).first.pseud_id).user }
      it "should have no errors" do
...

And

  describe "new" do
    let(:user) { Pseud.find(ChallengeSignup.in_collection(signup.collection).first.pseud_id).user }
    context "when prompt_type is offer" do

Removing the context block also fails in the same way

it errors with: 

  1) PromptsController new when prompt_type is offer should have no errors
     Failure/Error: expect(response).to have_http_status(200)
       expected the response to have status code 200 but it was 302
     # ./spec/controllers/prompts_controller_spec.rb:58:in `block (4 levels) in <top (required)>'

Which I think is not logging in properly